### PR TITLE
Add borderRadius property for rounded corners

### PR DESCRIPTION
**Pull request by hu9o**
Many websites/apps display profile pics with rounded corners.
This feature enables them to give users a more accurate preview of what the final image will look like.

Note that ``getImage()`` still produces a square-cornered image, without erasing the hidden corners.
This should be okay, as the corners are typically rounded on the webpage with CSS ``border-radius``.

### CODE REVIEW

**Comment by hu9o** [comment]
What do you people think ?

**Comment by mosch** [comment]
I think even if you like to display your images with rounded corners, you should still manage and keep the normal version of the images. 

I agree that the user should be able to get an accurate preview of the final image. I think you can add rounded corners to the editor by passing in roundedCorners via the style props:

    <AvatarEditor style={{borderRadius: 5}}>

somewhat like that?

**Comment by hu9o** [comment]
> I think even if you like to display your images with rounded corners, you should still manage and keep the normal version of the images.

That's how it's done: the output image doesn't change.

> ´´<AvatarEditor style={{borderRadius: 5}}>´´

This would round the corners of the canvas. The feature I propose allows to round the corners of the area inside the grey border instead, so that you can actually see what the final image with rounded corners will look like.

File: dist/index.js [hunk 1]
```diff
@@ -279,13 +281,30 @@
...
+            // clamp border radius between zero (perfect rectangle) and half the size without borders (perfect circle or "pill")
+            borderRadius = Math.max(borderRadius, 0);
+            borderRadius = Math.min(borderRadius, width / 2 - borderSize, height / 2 - borderSize);
+
+            context.beginPath();
+            // if the radius is zero, gotta worry no mo' : spare some cpu sweat by just drawing a rect
+            if (borderRadius === 0) {
+                context.rect(borderSize, borderSize, width - borderSize * 2, height - borderSize * 2);
+            } else {
+                var sizePlusRad = borderSize + borderRadius;
```
Comment (mosch) [inline dist/index.js]: Do you mind moving this code to a separate method just for organisation?

**Comment by hu9o** [comment]
A ``drawRoundedRectangle(context, x, y, width, height, radius)`` method, perhaps ?

Alternatively, I would like to add the ``roundedRect(x, y, width, height, radius)`` method to the canvas' context. The benefits: It would work just like ``context.rect()``, the intent would be very clear, and it would avoid adding another method (to AvatarEditor) that's only ever used in one place.

Problem is, I can't find a satisfying way to do that...

**Comment by mosch** [comment]
I like your `drawRoundedRectangle()` but I think there is no need to have it on the AvatarEditor. So put might want to put it somewere above or below.

**Comment by hu9o** [comment]
Here you go

**Comment by mosch** [comment]
Thanks for this, what do you think about adding documentation for this? :+1:

**Comment by hu9o** [comment]
Sure, I forgot about that

**Comment by hu9o** [comment]
Done !

**Pull request merged**